src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -102,6 +104,25 @@ class UserService(
     )
   }
 
+  private fun updateUserFromCommunityApi(user: UserEntity): UserEntity {
+    val staffUserDetailsResponse = communityApiClient.getStaffUserDetails(user.deliusUsername)
+
+    val staffUserDetails = when (staffUserDetailsResponse) {
+      is ClientResult.Success -> staffUserDetailsResponse.body
+      is ClientResult.Failure -> staffUserDetailsResponse.throwException()
+    }
+
+    user.apply {
+      name = "${staffUserDetails.staff.forenames} ${staffUserDetails.staff.surname}"
+      deliusStaffIdentifier = staffUserDetails.staffIdentifier
+      deliusStaffCode = staffUserDetails.staffCode
+      email = staffUserDetails.email
+      telephoneNumber = staffUserDetails.telephoneNumber
+    }
+
+    return userRepository.save(user)
+  }
+
```

Comment:
hmm,

I think we may be replicating some similar logic within the `getUserForId()` function?

https://github.com/ministryofjustice/approved-premises-api/blob/4023d11c9a5ce2a1b32824dacda85c3cc9bc4bc8/src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt#L47-L65

perhaps we need to refactor so that both `getUserForUsername()` and `getUserForId()` use a common `updateUserIfChanged()` function?

At any rate I think we'll need to ensure that the `staffCode` is updated in both pathways?

Reply:
Good point, I'm hesitant to combine these methods as getUserForId calls Community API every time whereas getUserForUsername just uses what's stored in the DB (and is called much more prolifically.)

Have updated getUserForId to also update staffCode though.

Reply:
👍 sounds good, thks!